### PR TITLE
feat: add config template and defaults for sqlite-only store

### PR DESCRIPTION
**Pull request by jm-clius**
This PR adds the config template and defaults for the SQLite store feature, available in nwaku `v0.10` and above.

Subsequent PRs will follow in infra-nim-waku and infra-status for fleet specific configuration.

cc @richard-ramos @kaiserd

### CODE REVIEW

**Review by kaiserd — APPROVED** [review]
LGTM!

**Review by richard-ramos — APPROVED** [review]

File: defaults/main.yml [hunk 1]
```diff
@@ -43,6 +43,10 @@ nim_waku_p2p_max_connections: 150
 # maximum number of stored historical messages
 nim_waku_store_capacity: 10000
 
+# SQLite store
+nim_waku_sqlite_store: false
+nim_waku_sqlite_retention_time: 30.days.seconds
```
Comment (jakubgs) [inline defaults/main.yml:48]: Could we explain a bit this weird duration format? Also, this text should be single-quoted.

Maybe we could also start with a lower default in the role? 7 days?

File: templates/docker-compose.yml.j2 [hunk 2]
```diff
@@ -50,6 +50,8 @@ services:
       --max-connections={{ nim_waku_p2p_max_connections }}
       --store-capacity={{ nim_waku_store_capacity }}
       --dns4-domain-name={{ nim_waku_dns4_domain_name }}
+      --sqlite-store={{ nim_waku_sqlite_store }}
+      --sqlite-retention-time={{ nim_waku_sqlite_retention_time }}
```
Comment (jakubgs) [inline templates/docker-compose.yml.j2:54]: If store is disabled the `--sqlite-retention-time` flag should be conditional.

Also, are we sure the name of the flag should mention a specific database? Wouldn't it make more sense for the name of the flag to be db-agnostic? Like `--database-retention-time` or `--db-retention-time`?

**Comment by jm-clius** [comment]
@jakubgs I have addressed your comments. If you're happy, we can rebase, squash and merge.

**Review by jakubgs — APPROVED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (jm-clius) [reply defaults/main.yml]: I think this is simply an "easy" way to define "30 days worth of seconds" in chronos. Agreed re lowering the default here.

File: templates/docker-compose.yml.j2 (hunk 2 above)
Comment (jm-clius) [reply templates/docker-compose.yml.j2:55]: Good point re the naming. My understanding is that we want to consolidate (and significantly simplify!) store config when we define a general store interface which should be agnostic to the specific store implementation behind it (whether in-memory, any DB technology, or a hybrid). For now the sqlite store is a beta feature behind a very specific feature flag to allow us to test the concept. @kaiserd, any thoughts?

Comment (kaiserd) [reply templates/docker-compose.yml.j2:55]: Agree with @jm-clius . I would suggest making this part of the PR that introduces the new store interface.
Also, we plan to introduce subcommands soon, which allows making flags like this conditional.

File: defaults/main.yml (hunk 1 above)
Comment (kaiserd) [reply defaults/main.yml]: The idea was to allow a chronos duration as type for the parameter. But it is not supported out of the box, so I went with seconds for now.

**Pull request merged**